node/pkg/chain/helper/helper.go

```diff
@@ -236,11 +244,79 @@ func (t *ChainHelper) PublicAddressString() (string, error) {
 	return address.Hex(), nil
 }
 
+func (t *ChainHelper) SubmitDirect(ctx context.Context, contractAddress string, functionString string, args ...interface{}) error {
+	tx, err := t.MakeDirectTx(ctx, contractAddress, functionString, args...)
+	if err != nil {
+		return err
+	}
+
+	for _, client := range t.clients {
+		err := utils.SubmitRawTx(ctx, client, tx)
+		if err != nil {
+			if utils.ShouldRetryWithSwitchedJsonRPC(err) {
+				log.Error().Err(err).Msg("Error on retrying on JsonRpcFailure")
+				continue
+			}
+
+			if utils.IsNonceError(err) || utils.IsNonceAlreadyInPool(err) {
+				log.Error().Err(err).Msg("Error on retrying on NonceFailure")
+				tx, err = t.MakeDirectTx(ctx, contractAddress, functionString, args...)
+				if err != nil {
+					return err
+				}
+				continue
+			}
+			return err
+		}
+		break
+	}
+
+	return nil
+}
+
+func (t *ChainHelper) SubmitDelegated(ctx context.Context, contractAddress string, functionString string, args ...interface{}) error {
+	tx, err := t.MakeFeeDelegatedTx(ctx, contractAddress, functionString, args...)
+	if err != nil {
+		return err
+	}
+
+	tx, err = t.GetSignedFromDelegator(tx)
+	if err != nil {
+		return err
+	}
+
+	for _, client := range t.clients {
+		err := utils.SubmitRawTx(ctx, client, tx)
+		if err != nil {
+			if utils.ShouldRetryWithSwitchedJsonRPC(err) {
+				log.Error().Err(err).Msg("Error on retrying on JsonRpcFailure")
+				continue
+			}
+			if utils.IsNonceError(err) || utils.IsNonceAlreadyInPool(err) {
+				log.Error().Err(err).Msg("Error on retrying on NonceFailure")
+				tx, err = t.MakeFeeDelegatedTx(ctx, contractAddress, functionString, args...)
+				if err != nil {
+					return err
+				}
+				tx, err = t.GetSignedFromDelegator(tx)
+				if err != nil {
+					return err
+				}
+				continue
```

Comment:
It doesn't have to be part of this PR but would be good to use the same json-rpc in case of nonce error. Here is an edge case scenario:

If we provide only a single json rpc and we get nonce error, the loop will exit (since the clients length is 1) and the failed `tx` won't be retried

Reply:
awh I see, I'll see what I can do 👍

Reply:
made an update for this, please double check the logic! 🙏